syntax = "proto3";

import "google/protobuf/empty.proto";

// Servicer for rpc agent server
service RpcAgent {
    rpc is_alive (google.protobuf.Empty) returns (StatusResponse) {}
    rpc call_func (RpcMsg) returns (RpcMsg) {}
}

// Message class for rpc communication
message RpcMsg {
    string value = 1;
    string target_func = 2;
    string agent_id = 3;
}

message StatusResponse {
    bool ok = 1;
    string message = 2;
}